spec/operators/partition-spec.js

```diff
@@ -1,78 +1,81 @@
-      return value >= 0;
-    });
+    expectObservableArray(e1.partition(predicate), expected);
+  });
 
-    var positiveStream = streams[0];
-    var negativeStream = streams[1];
+  it('should pass errors to both returned observables', function () {
+    function predicate(x) {
+      return x === 'a';
+    }
+    var e1 =    hot('--a-b---#');
+    var expected = ['--a-----#',
+                    '----b---#'];
 
-    positiveValues = [];
-    negativeValues = [];
+    expectObservableArray(e1.partition(predicate), expected);
+  });
 
-    positiveStream.subscribe(function (value) {
-      positiveValues.push(value);
-    }, null, completer);
+  it('should partition empty observable if source does not emits', function () {
+    function predicate(x) {
+      return x === 'x';
+    }
+    var e1 =    hot('----|');
+    var expected = ['----|',
+                    '----|'];
 
-    negativeStream.subscribe(function (value) {
-      negativeValues.push(value);
-    }, null, completer);
+    expectObservableArray(e1.partition(predicate), expected);
   });
 
-  it('should pass errors to both returned observables', function (done) {
-    var values = [-3, -2, -1, 0, 1, 2];
-    var numberStream = Rx.Observable.fromArray(values);
-    var errored = false;
-
-    function rejecter() {
-      expect(true).toBe(false);
+  it('should partition if source emits single elements', function () {
+    function predicate(x) {
+      return x === 'a';
     }
+    var e1 =    hot('--a--|');
+    var expected = ['--a--|',
+                    '-----|'];
+
+    expectObservableArray(e1.partition(predicate), expected);
+  });
 
-    function completer(error) {
-      if (!errored) {
-        errored = error;
-      } else {
-        expect(errored).toBe(error);
-        done();
-      }
+  it('should partition to infinite observable if source does not completes', function () {
+    function predicate(x) {
+      return x === 'a';
     }
+    var e1 =    hot('--a-b---a------d----');
+    var expected = ['--a-----a-',
+                    '----b----------d-'];
 
-    var streams = numberStream
-      .map(function (value) {
-        throw 'error';
-      })
-      .partition(function (value) {
-        return value >= 0;
-      });
+    expectObservableArray(e1.partition(predicate), expected);
+  });
 
-    var positiveStream = streams[0];
-    var negativeStream = streams[1];
+  it('should partition into two observable with early unsubscription', function () {
+    function predicate(x) {
+      return x === 'a';
+    }
+    var e1 =    hot('--a-b---a------d-|');
+    var unsub =     '-------!';
+    var expected = ['--a-',
+                    '----b-'];
+    var result = e1.partition(predicate);
 
-    positiveStream.subscribe(rejecter, completer, rejecter);
-    negativeStream.subscribe(rejecter, completer, rejecter);
+    for (var idx = 0; idx < result.length; idx++ ) {
+      expectObservable(result[idx], unsub).toBe(expected[idx]);
+    }
```

Comment:
This test case is only case accepting unsubscription as parameter, choose simplify `expectObservableArray` and let loop explicitly with unsubscription.